Development/pcm_alt.yml

```diff
@@ -0,0 +1,45 @@
+name: ConsIndShock
+
+states:
+  - name: m
+    long_name: market resources
+    domain: [0, .inf]
+
+controls:
+  - name: c
+    long_name: consumption
+    domain: [0, .inf]
+  - name: stigma
+    long_name: risky portfolio share
+    domain: [0, 1]
+
+distributions:
+  perm: !MeanOneLogNormal
+    std: std_perm
+  tran: !MeanOneLogNormal
+    std: std_tran
+  risky: !LogNormal
+    mean: mean_risky
+    std: std_risky
+
+parameters:
+  beta: 0.96
+  rho: 2
+  R: 1.04
+  G: 1.02
+  std_perm: 0.1
+  std_tran: 0.1
+  mean_risky: 1.08
+  std_risky: 0.2
+
+equations: # section containing the list of equations
+  transitions:
+    m_: R_port * a / (G * perm) + tran
+    v(m): u(c) + beta * E[(G * perm_)^(1-rho) * v(m_)]

```

Comment:
is the value function also related to a solution concept?

Bellman equations, Euler equations, and discounted value are all conditions that can be used by a solver to 'solve' a model. The 'solution' really is the the optimal policy.

(See Malier, Malier, and Winant, 2021)

Reply:
The value function describes the problem generally, so it's related to any solution concept. We technically wouldn't need anything else to solve a problem in a brute force way, just max the value function. Other operators (EGM, Policy function iteration, Euler equation) are used by different solution methods, but they are not essential to describing the problem that needs to be solved. 

In defining the recursive problem, we need to define the value function. Some equations might be superfluous, but perhaps there are situations where other equations are actually required to describe the problem.

Reply:
I'm not sure I'm convinced by this.

I like your syntax around `aggregator` that you outline here, better: https://github.com/econ-ark/OverARK/pull/46#discussion_r1369204888

Reply:
In SolvingMicroDSOPs, Sections 2 and 3 define the problem using the value function. This is how I generally see problems described in Econ, the optional first step is to define the sequential problem, then generally you see the recursive problem. It's not until you start talking about how to solve the problem that you start taking first order conditions (Euler, etc). 

https://llorracc.github.io/SolvingMicroDSOPs/

Reply:
I understand that this is how Chris does it.

But in Maliar, Maliar, and Winant 2021, section 2, they take a different approach, and survey a wider span of literature at the bottom of page 81: https://web.stanford.edu/~maliars/Files/JME2021.pdf

Here, they define lifetime reward first, as the general nature of the optimization problem:

Reply:
Equation 4 is what is known as the sequential problem. Dynamic programming broadly consists of recasting the sequential problem into a recursive problem, along with the conditions necessary for the solution to the recursive problem to also be the solution for the sequential problem. This last step is very abstract and theoretical, and a lot of papers don't bother with this. So, they start with the recursive problem defined by the value function. 

Additionally, in page 81, you'll notice that the FOC and the Envelope condition are derivatives of the value function, so if only the value function was provided, we could always generate these conditions. 

I agree that in general, the sequential problem is what we are trying to solve, so it's the most important description of the problem. However, we are also specifically solving the problem using Dynamic Programming (recursive formulation), which may have different methods available, but nevertheless all derive from the recursive value function.

Reply:
I agree with your characterization of dynamic programming.
I think that for all solution methods derived from dynamic programming, it makes sense to include an `aggregator` statement.
The value function, its derivatives, etc. can go there.

But what about solution methods that are not based on dynamic programming? For example:
- projection methods ( see https://people.duke.edu/~acb8/326/proj.pdf https://kenjudd.org/wp-content/uploads/2020/08/Lecture_19_1-Projection_Methods.pdf ) 
- The MMW'21 method

In MMW'21, they demonstrate that the model can be solved using lifetime reward directly, without using recursive form. Its performance is comparable to the Euler equation method. See Section 4.3

Reply:
I see, I understand MMW'21 a bit better now. Thank you. 

I had not considered non recursive approaches, mostly because I'm not very familiar with them.